tsParticles 1.18.0

matteobruni opened the pull request:
# tsParticles 1.18.0

## Release Notes

### New Features

- Manual Particles (closes #839)
- Links and Triangles frequency (closes #691)
- Mouse hover can act as a light source (closes #606)
- Particles duplicates can be reduced (closes #437)
- Options have now a dedicated GUI editor library (closes #534)
- Theming options (closes #646)
- Added support to HSV colors (closes #692)
- OutMode can be configured for every canvas edge (closes #739)
- Bounce interaction mode; divs and mouse hover have this mode (closes #774)
- Particles can have a max radius to move inside (closes #797)
- Particles movement can be affected by the size, the higher the size, the higher the speed
- Particles movement has now gravity options
- Particles can have a custom life span and a life count, like the emitter options
- Particles can now configure bounces on canvas edge and for particles collisions
- New Motion settings for `prefer-reduced-motion` users, it can reduce or disable animations

### Bug Fixes

- Single exported options type, used by all components (closes #898)
- Fixed click bubble mode (closes #618)

### Breaking Changes

- Color animation with `sync` false in 1.17.0 used a random color instead of a random speed. This is fixed in 1.18.0 but you need to use the `random` value in color and not `sync: false` to achieve the same effect.

dependabot-preview[bot] committed:
build(deps-dev): bump @angular-devkit/build-angular

Bumps [@angular-devkit/build-angular](https://github.com/angular/angular-cli) from 0.1001.0 to 0.1001.1.
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Commits](https://github.com/angular/angular-cli/commits)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump rollup-plugin-svelte from 6.0.0 to 6.0.1

Bumps [rollup-plugin-svelte](https://github.com/sveltejs/rollup-plugin-svelte) from 6.0.0 to 6.0.1.
- [Release notes](https://github.com/sveltejs/rollup-plugin-svelte/releases)
- [Changelog](https://github.com/sveltejs/rollup-plugin-svelte/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sveltejs/rollup-plugin-svelte/compare/v6.0.0...v6.0.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #820 from matteobruni/dependabot/npm_and_yarn/dev/eslint-7.9.0

build(deps-dev): bump eslint from 7.7.0 to 7.9.0

matteobruni committed:
Merge pull request #821 from matteobruni/dependabot/npm_and_yarn/dev/angular-devkit/build-angular-0.1001.1

build(deps-dev): bump @angular-devkit/build-angular from 0.1001.0 to 0.1001.1

matteobruni committed:
Merge pull request #822 from matteobruni/dependabot/npm_and_yarn/dev/rollup-plugin-svelte-6.0.1

build(deps-dev): bump rollup-plugin-svelte from 6.0.0 to 6.0.1

dependabot-preview[bot] committed:
build(deps-dev): bump @vue/cli-service from 4.5.4 to 4.5.6

Bumps [@vue/cli-service](https://github.com/vuejs/vue-cli/tree/HEAD/packages/@vue/cli-service) from 4.5.4 to 4.5.6.
- [Release notes](https://github.com/vuejs/vue-cli/releases)
- [Changelog](https://github.com/vuejs/vue-cli/blob/dev/CHANGELOG.md)
- [Commits](https://github.com/vuejs/vue-cli/commits/v4.5.6/packages/@vue/cli-service)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump @angular/platform-browser from 10.1.0 to 10.1.1

Bumps [@angular/platform-browser](https://github.com/angular/angular/tree/HEAD/packages/platform-browser) from 10.1.0 to 10.1.1.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/10.1.1/packages/platform-browser)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump @commitlint/config-conventional

Bumps [@commitlint/config-conventional](https://github.com/conventional-changelog/commitlint) from 9.1.2 to 11.0.0.
- [Release notes](https://github.com/conventional-changelog/commitlint/releases)
- [Changelog](https://github.com/conventional-changelog/commitlint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/conventional-changelog/commitlint/compare/v9.1.2...v11.0.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #824 from matteobruni/dependabot/npm_and_yarn/dev/vue/cli-service-4.5.6

build(deps-dev): bump @vue/cli-service from 4.5.4 to 4.5.6

matteobruni committed:
Merge pull request #825 from matteobruni/dependabot/npm_and_yarn/dev/angular/platform-browser-10.1.1

build(deps): bump @angular/platform-browser from 10.1.0 to 10.1.1

matteobruni committed:
Merge pull request #826 from matteobruni/dependabot/npm_and_yarn/dev/commitlint/config-conventional-11.0.0

build(deps-dev): bump @commitlint/config-conventional from 9.1.2 to 11.0.0

matteobruni committed:
Merge branch 'update-docs' into dev

# Conflicts:
#	core/main/src/Core/Container.ts
#	core/main/src/Core/FrameManager.ts
#	core/main/src/pjs.ts

matteobruni committed:
build: angular updates

dependabot-preview[bot] committed:
build(deps-dev): bump svelte-check from 1.0.38 to 1.0.42

Bumps [svelte-check](https://github.com/sveltejs/language-tools) from 1.0.38 to 1.0.42.
- [Release notes](https://github.com/sveltejs/language-tools/releases)
- [Commits](https://github.com/sveltejs/language-tools/commits)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump typedoc from 0.19.0 to 0.19.1

Bumps [typedoc](https://github.com/TypeStrong/TypeDoc) from 0.19.0 to 0.19.1.
- [Release notes](https://github.com/TypeStrong/TypeDoc/releases)
- [Commits](https://github.com/TypeStrong/TypeDoc/compare/v0.19.0...v0.19.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump jsoneditor from 9.0.5 to 9.1.0

Bumps [jsoneditor](https://github.com/josdejong/jsoneditor) from 9.0.5 to 9.1.0.
- [Release notes](https://github.com/josdejong/jsoneditor/releases)
- [Changelog](https://github.com/josdejong/jsoneditor/blob/develop/HISTORY.md)
- [Commits](https://github.com/josdejong/jsoneditor/compare/v9.0.5...v9.1.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #827 from matteobruni/dependabot/npm_and_yarn/dev/svelte-check-1.0.42

build(deps-dev): bump svelte-check from 1.0.38 to 1.0.42

matteobruni committed:
Merge pull request #828 from matteobruni/dependabot/npm_and_yarn/dev/typedoc-0.19.1

build(deps-dev): bump typedoc from 0.19.0 to 0.19.1

matteobruni committed:
Merge pull request #829 from matteobruni/dependabot/npm_and_yarn/dev/jsoneditor-9.1.0

build(deps-dev): bump jsoneditor from 9.0.5 to 9.1.0

matteobruni committed:
Merge branch 'update-docs' into dev

dependabot-preview[bot] committed:
build(deps-dev): bump svelte from 3.25.0 to 3.25.1

Bumps [svelte](https://github.com/sveltejs/svelte) from 3.25.0 to 3.25.1.
- [Release notes](https://github.com/sveltejs/svelte/releases)
- [Changelog](https://github.com/sveltejs/svelte/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sveltejs/svelte/compare/v3.25.0...v3.25.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump prettier from 2.1.1 to 2.1.2

Bumps [prettier](https://github.com/prettier/prettier) from 2.1.1 to 2.1.2.
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/2.1.1...2.1.2)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump webpack-bundle-analyzer from 3.8.0 to 3.9.0

Bumps [webpack-bundle-analyzer](https://github.com/webpack-contrib/webpack-bundle-analyzer) from 3.8.0 to 3.9.0.
- [Release notes](https://github.com/webpack-contrib/webpack-bundle-analyzer/releases)
- [Changelog](https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/webpack-bundle-analyzer/commits)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #833 from matteobruni/dependabot/npm_and_yarn/dev/svelte-3.25.1

build(deps-dev): bump svelte from 3.25.0 to 3.25.1

matteobruni committed:
Merge pull request #834 from matteobruni/dependabot/npm_and_yarn/dev/prettier-2.1.2

build(deps-dev): bump prettier from 2.1.1 to 2.1.2

matteobruni committed:
Merge pull request #835 from matteobruni/dependabot/npm_and_yarn/dev/webpack-bundle-analyzer-3.9.0

build(deps-dev): bump webpack-bundle-analyzer from 3.8.0 to 3.9.0

matteobruni committed:
Merge branch 'update-docs' into dev

matteobruni committed:
Merge branch 'update-docs' into dev

matteobruni committed:
Merge branch 'update-docs' into dev

# Conflicts:
#	core/main/package.json
#	yarn.lock

matteobruni committed:
build: fixed preact build

dependabot-preview[bot] committed:
build(deps-dev): bump sass from 1.26.10 to 1.26.11

Bumps [sass](https://github.com/sass/dart-sass) from 1.26.10 to 1.26.11.
- [Release notes](https://github.com/sass/dart-sass/releases)
- [Changelog](https://github.com/sass/dart-sass/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sass/dart-sass/compare/1.26.10...1.26.11)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #840 from matteobruni/dependabot/npm_and_yarn/dev/sass-1.26.11

build(deps-dev): bump sass from 1.26.10 to 1.26.11

dependabot-preview[bot] committed:
build(deps-dev): bump @rollup/plugin-commonjs from 15.0.0 to 15.1.0

Bumps [@rollup/plugin-commonjs](https://github.com/rollup/plugins) from 15.0.0 to 15.1.0.
- [Release notes](https://github.com/rollup/plugins/releases)
- [Commits](https://github.com/rollup/plugins/compare/commonjs-v15.0.0...commonjs-v15.1.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #841 from matteobruni/dependabot/npm_and_yarn/dev/rollup/plugin-commonjs-15.1.0

build(deps-dev): bump @rollup/plugin-commonjs from 15.0.0 to 15.1.0

dependabot-preview[bot] committed:
build(deps-dev): bump @vue/cli-plugin-typescript from 4.5.4 to 4.5.6

Bumps [@vue/cli-plugin-typescript](https://github.com/vuejs/vue-cli/tree/HEAD/packages/@vue/cli-plugin-typescript) from 4.5.4 to 4.5.6.
- [Release notes](https://github.com/vuejs/vue-cli/releases)
- [Changelog](https://github.com/vuejs/vue-cli/blob/dev/CHANGELOG.md)
- [Commits](https://github.com/vuejs/vue-cli/commits/v4.5.6/packages/@vue/cli-plugin-typescript)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #842 from matteobruni/dependabot/npm_and_yarn/dev/vue/cli-plugin-typescript-4.5.6

build(deps-dev): bump @vue/cli-plugin-typescript from 4.5.4 to 4.5.6

dependabot-preview[bot] committed:
build(deps-dev): bump postcss from 7.0.32 to 8.0.6

Bumps [postcss](https://github.com/postcss/postcss) from 7.0.32 to 8.0.6.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/master/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/7.0.32...8.0.6)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump @angular-devkit/build-angular

Bumps [@angular-devkit/build-angular](https://github.com/angular/angular-cli) from 0.1001.1 to 0.1001.2.
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Commits](https://github.com/angular/angular-cli/commits)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #843 from matteobruni/dependabot/npm_and_yarn/dev/angular-devkit/build-angular-0.1001.2

build(deps-dev): bump @angular-devkit/build-angular from 0.1001.1 to 0.1001.2

dependabot-preview[bot] committed:
build(deps-dev): bump svelte-preprocess from 4.2.1 to 4.3.0

Bumps [svelte-preprocess](https://github.com/sveltejs/svelte-preprocess) from 4.2.1 to 4.3.0.
- [Release notes](https://github.com/sveltejs/svelte-preprocess/releases)
- [Changelog](https://github.com/sveltejs/svelte-preprocess/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sveltejs/svelte-preprocess/compare/v4.2.1...v4.3.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #844 from matteobruni/dependabot/npm_and_yarn/dev/svelte-preprocess-4.3.0

build(deps-dev): bump svelte-preprocess from 4.2.1 to 4.3.0

dependabot-preview[bot] committed:
build(deps-dev): bump svelte-check from 1.0.42 to 1.0.46

Bumps [svelte-check](https://github.com/sveltejs/language-tools) from 1.0.42 to 1.0.46.
- [Release notes](https://github.com/sveltejs/language-tools/releases)
- [Commits](https://github.com/sveltejs/language-tools/commits)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump vue-class-component from 7.2.5 to 7.2.6

Bumps [vue-class-component](https://github.com/vuejs/vue-class-component) from 7.2.5 to 7.2.6.
- [Release notes](https://github.com/vuejs/vue-class-component/releases)
- [Commits](https://github.com/vuejs/vue-class-component/compare/v7.2.5...v7.2.6)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #846 from matteobruni/dependabot/npm_and_yarn/dev/vue-class-component-7.2.6

build(deps): bump vue-class-component from 7.2.5 to 7.2.6

matteobruni committed:
Merge pull request #847 from matteobruni/dependabot/npm_and_yarn/dev/svelte-check-1.0.46

build(deps-dev): bump svelte-check from 1.0.42 to 1.0.46

matteobruni committed:
Merge pull request #849 from matteobruni/dependabot/npm_and_yarn/dev/postcss-8.0.6

build(deps-dev): bump postcss from 7.0.32 to 8.0.6

dependabot-preview[bot] committed:
build(deps-dev): bump @commitlint/cli from 9.1.2 to 11.0.0

Bumps [@commitlint/cli](https://github.com/conventional-changelog/commitlint) from 9.1.2 to 11.0.0.
- [Release notes](https://github.com/conventional-changelog/commitlint/releases)
- [Changelog](https://github.com/conventional-changelog/commitlint/blob/master/CHANGELOG.md)
- [Commits](https://github.com/conventional-changelog/commitlint/compare/v9.1.2...v11.0.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #848 from matteobruni/dependabot/npm_and_yarn/dev/commitlint/cli-11.0.0

build(deps-dev): bump @commitlint/cli from 9.1.2 to 11.0.0

matteobruni committed:
Merge branch 'master' into dev

matteobruni committed:
build: updates

matteobruni committed:
Merge branch 'master' into dev

matteobruni committed:
Merge branch 'master' into dev

matteobruni committed:
feat: found a good solution with good performance to fix #691

matteobruni committed:
feat: added movement speed affected by particles size

matteobruni committed:
fix: fixed bounce for #739

matteobruni committed:
build: updated test preset

matteobruni committed:
build: cleaned some console.logs

matteobruni committed:
feat: added HSV support

matteobruni committed:
feat: completed HSV support

matteobruni committed:
fix: editor fixed with new color refactoring

dependabot-preview[bot] committed:
build(deps-dev): bump postcss from 8.0.6 to 8.0.7

Bumps [postcss](https://github.com/postcss/postcss) from 8.0.6 to 8.0.7.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/master/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.0.6...8.0.7)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump svelte from 3.25.1 to 3.26.0

Bumps [svelte](https://github.com/sveltejs/svelte) from 3.25.1 to 3.26.0.
- [Release notes](https://github.com/sveltejs/svelte/releases)
- [Changelog](https://github.com/sveltejs/svelte/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sveltejs/svelte/compare/v3.25.1...v3.26.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump rollup from 2.26.11 to 2.28.1

Bumps [rollup](https://github.com/rollup/rollup) from 2.26.11 to 2.28.1.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v2.26.11...v2.28.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #853 from matteobruni/dependabot/npm_and_yarn/dev/postcss-8.0.7

build(deps-dev): bump postcss from 8.0.6 to 8.0.7

matteobruni committed:
Merge pull request #855 from matteobruni/dependabot/npm_and_yarn/dev/svelte-3.26.0

build(deps-dev): bump svelte from 3.25.1 to 3.26.0

matteobruni committed:
Merge pull request #856 from matteobruni/dependabot/npm_and_yarn/dev/rollup-2.28.1

build(deps-dev): bump rollup from 2.26.11 to 2.28.1

matteobruni committed:
build: updated packages

matteobruni committed:
fix: fixed #618

matteobruni committed:
build: renamed interaction mouse folder

matteobruni committed:
feat: bounce mode for divs, working only on squared divs but it's something

matteobruni committed:
feat: bounce mode is now working fine for circular divs too
next step is clean code, there's a lot of duplicated code for collisions
maybe an utils class could prevent this

matteobruni committed:
feat: added sides count to particle to improve the shadow for the light over

matteobruni committed:
docs: added wiki pages for Themes and Background Mode

matteobruni committed:
docs: added set theme method in container documentation

matteobruni committed:
docs: add some typedoc comments to some methods

matteobruni committed:
build: updated object-gui and updated color editors, added some new properties

dependabot-preview[bot] committed:
build(deps): bump @angular/common from 10.1.2 to 10.1.3

Bumps [@angular/common](https://github.com/angular/angular/tree/HEAD/packages/common) from 10.1.2 to 10.1.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/10.1.3/packages/common)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump postcss from 8.0.7 to 8.0.9

Bumps [postcss](https://github.com/postcss/postcss) from 8.0.7 to 8.0.9.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/master/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.0.7...8.0.9)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump @angular/compiler-cli from 10.1.2 to 10.1.3

Bumps [@angular/compiler-cli](https://github.com/angular/angular/tree/HEAD/packages/compiler-cli) from 10.1.2 to 10.1.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/10.1.3/packages/compiler-cli)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump @angular/platform-browser from 10.1.2 to 10.1.3

Bumps [@angular/platform-browser](https://github.com/angular/angular/tree/HEAD/packages/platform-browser) from 10.1.2 to 10.1.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/10.1.3/packages/platform-browser)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump preact from 10.4.8 to 10.5.2

Bumps [preact](https://github.com/preactjs/preact) from 10.4.8 to 10.5.2.
- [Release notes](https://github.com/preactjs/preact/releases)
- [Commits](https://github.com/preactjs/preact/compare/10.4.8...10.5.2)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #866 from matteobruni/dependabot/npm_and_yarn/dev/preact-10.5.2

build(deps): bump preact from 10.4.8 to 10.5.2

matteobruni committed:
Merge pull request #862 from matteobruni/dependabot/npm_and_yarn/dev/postcss-8.0.9

build(deps-dev): bump postcss from 8.0.7 to 8.0.9

matteobruni committed:
Merge pull request #858 from matteobruni/dependabot/npm_and_yarn/dev/angular/common-10.1.3

build(deps): bump @angular/common from 10.1.2 to 10.1.3

matteobruni committed:
Merge pull request #863 from matteobruni/dependabot/npm_and_yarn/dev/angular/compiler-cli-10.1.3

build(deps-dev): bump @angular/compiler-cli from 10.1.2 to 10.1.3

matteobruni committed:
Merge pull request #864 from matteobruni/dependabot/npm_and_yarn/dev/angular/platform-browser-10.1.3

build(deps): bump @angular/platform-browser from 10.1.2 to 10.1.3

dependabot-preview[bot] committed:
build(deps): bump @angular/compiler from 10.1.2 to 10.1.3

Bumps [@angular/compiler](https://github.com/angular/angular/tree/HEAD/packages/compiler) from 10.1.2 to 10.1.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/10.1.3/packages/compiler)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump @angular/cli from 10.1.2 to 10.1.3

Bumps [@angular/cli](https://github.com/angular/angular-cli) from 10.1.2 to 10.1.3.
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Commits](https://github.com/angular/angular-cli/compare/v10.1.2...v10.1.3)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump @angular/animations from 10.1.2 to 10.1.3

Bumps [@angular/animations](https://github.com/angular/angular/tree/HEAD/packages/animations) from 10.1.2 to 10.1.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/10.1.3/packages/animations)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #859 from matteobruni/dependabot/npm_and_yarn/dev/angular/compiler-10.1.3

build(deps): bump @angular/compiler from 10.1.2 to 10.1.3

matteobruni committed:
Merge pull request #861 from matteobruni/dependabot/npm_and_yarn/dev/angular/animations-10.1.3

build(deps): bump @angular/animations from 10.1.2 to 10.1.3

matteobruni committed:
Merge pull request #865 from matteobruni/dependabot/npm_and_yarn/dev/angular/cli-10.1.3

build(deps-dev): bump @angular/cli from 10.1.2 to 10.1.3

matteobruni committed:
feat: fixed some editor outdated fields and added some new

matteobruni committed:
fix: fixed editor properties, missing themes for now

matteobruni committed:
build(deps): updated object-gui

dependabot-preview[bot] committed:
build(deps): bump @angular/router from 10.1.2 to 10.1.3

Bumps [@angular/router](https://github.com/angular/angular/tree/HEAD/packages/router) from 10.1.2 to 10.1.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/10.1.3/packages/router)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump svelte-check from 1.0.47 to 1.0.49

Bumps [svelte-check](https://github.com/sveltejs/language-tools) from 1.0.47 to 1.0.49.
- [Release notes](https://github.com/sveltejs/language-tools/releases)
- [Commits](https://github.com/sveltejs/language-tools/commits)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump @angular/core from 10.1.2 to 10.1.3

Bumps [@angular/core](https://github.com/angular/angular/tree/HEAD/packages/core) from 10.1.2 to 10.1.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/10.1.3/packages/core)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump @angular-devkit/build-angular

Bumps [@angular-devkit/build-angular](https://github.com/angular/angular-cli) from 0.1001.2 to 0.1001.3.
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Commits](https://github.com/angular/angular-cli/commits)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

snyk-bot committed:
fix: upgrade zone.js from 0.10.3 to 0.11.1

Snyk has created this PR to upgrade zone.js from 0.10.3 to 0.11.1.

See this package in npm:
https://www.npmjs.com/package/zone.js

See this project in Snyk:
https://app.snyk.io/org/matteobruni/project/b76ca069-8e30-4dd1-8164-cd97ed52066f?utm_source=github&utm_medium=upgrade-pr

snyk-bot committed:
fix: upgrade multiple dependencies with Snyk

Snyk has created this PR to upgrade:
  - @angular/animations from 10.0.14 to 10.1.0.
    See this package in npm: https://www.npmjs.com/package/@angular/animations
  - @angular/common from 10.0.14 to 10.1.0.
    See this package in npm: https://www.npmjs.com/package/@angular/common
  - @angular/compiler from 10.0.14 to 10.1.0.
    See this package in npm: https://www.npmjs.com/package/@angular/compiler
  - @angular/core from 10.0.14 to 10.1.0.
    See this package in npm: https://www.npmjs.com/package/@angular/core
  - @angular/forms from 10.0.14 to 10.1.0.
    See this package in npm: https://www.npmjs.com/package/@angular/forms
  - @angular/platform-browser from 10.0.14 to 10.1.0.
    See this package in npm: https://www.npmjs.com/package/@angular/platform-browser
  - @angular/platform-browser-dynamic from 10.0.14 to 10.1.0.
    See this package in npm: https://www.npmjs.com/package/@angular/platform-browser-dynamic
  - @angular/router from 10.0.14 to 10.1.0.
    See this package in npm: https://www.npmjs.com/package/@angular/router

See this project in Snyk:
https://app.snyk.io/org/matteobruni/project/37d57f13-d548-468a-8d72-a84e4b12d4d4?utm_source=github&utm_medium=upgrade-pr

matteobruni committed:
Merge pull request #867 from matteobruni/dependabot/npm_and_yarn/dev/angular/router-10.1.3

build(deps): bump @angular/router from 10.1.2 to 10.1.3

matteobruni committed:
Merge pull request #868 from matteobruni/dependabot/npm_and_yarn/dev/svelte-check-1.0.49

build(deps-dev): bump svelte-check from 1.0.47 to 1.0.49

matteobruni committed:
Merge pull request #869 from matteobruni/dependabot/npm_and_yarn/dev/angular/core-10.1.3

build(deps): bump @angular/core from 10.1.2 to 10.1.3

matteobruni committed:
Merge pull request #874 from matteobruni/dependabot/npm_and_yarn/dev/angular-devkit/build-angular-0.1001.3

build(deps-dev): bump @angular-devkit/build-angular from 0.1001.2 to 0.1001.3

dependabot-preview[bot] committed:
build(deps-dev): bump svelte from 3.26.0 to 3.27.0

Bumps [svelte](https://github.com/sveltejs/svelte) from 3.26.0 to 3.27.0.
- [Release notes](https://github.com/sveltejs/svelte/releases)
- [Changelog](https://github.com/sveltejs/svelte/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sveltejs/svelte/compare/v3.26.0...v3.27.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump @angular/platform-browser-dynamic

Bumps [@angular/platform-browser-dynamic](https://github.com/angular/angular/tree/HEAD/packages/platform-browser-dynamic) from 10.1.2 to 10.1.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/10.1.3/packages/platform-browser-dynamic)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge branch 'dev' into snyk-upgrade-6928c0f9399b13a9b7e2fb55a9be9d4e

matteobruni committed:
Merge branch 'dev' into snyk-upgrade-479d2bd7705298e0037a81a3ed1f3d8f

matteobruni committed:
Merge pull request #872 from matteobruni/dependabot/npm_and_yarn/dev/svelte-3.27.0

build(deps-dev): bump svelte from 3.26.0 to 3.27.0

matteobruni committed:
Merge pull request #870 from matteobruni/dependabot/npm_and_yarn/dev/angular/platform-browser-dynamic-10.1.3

build(deps): bump @angular/platform-browser-dynamic from 10.1.2 to 10.1.3

matteobruni committed:
Merge pull request #876 from matteobruni/snyk-upgrade-479d2bd7705298e0037a81a3ed1f3d8f

[Snyk] Upgrade: @angular/animations, @angular/common, @angular/compiler, @angular/core, @angular/forms, @angular/platform-browser, @angular/platform-browser-dynamic, @angular/router

matteobruni committed:
Merge pull request #875 from matteobruni/snyk-upgrade-6928c0f9399b13a9b7e2fb55a9be9d4e

[Snyk] Upgrade zone.js from 0.10.3 to 0.11.1

matteobruni committed:
build: updates

matteobruni committed:
feat: loadJSON can accept also a string array and an optional index parameter

matteobruni committed:
build: cleaned imports in editor

dependabot-preview[bot] committed:
build(deps-dev): bump eslint-plugin-vue

Bumps [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) from 7.0.0-beta.3 to 7.0.0-beta.4.
- [Release notes](https://github.com/vuejs/eslint-plugin-vue/releases)
- [Commits](https://github.com/vuejs/eslint-plugin-vue/compare/v7.0.0-beta.3...v7.0.0-beta.4)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump svelte-preprocess from 4.3.0 to 4.3.2

Bumps [svelte-preprocess](https://github.com/sveltejs/svelte-preprocess) from 4.3.0 to 4.3.2.
- [Release notes](https://github.com/sveltejs/svelte-preprocess/releases)
- [Changelog](https://github.com/sveltejs/svelte-preprocess/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sveltejs/svelte-preprocess/compare/v4.3.0...v4.3.2)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump jsoneditor from 9.1.0 to 9.1.1

Bumps [jsoneditor](https://github.com/josdejong/jsoneditor) from 9.1.0 to 9.1.1.
- [Release notes](https://github.com/josdejong/jsoneditor/releases)
- [Changelog](https://github.com/josdejong/jsoneditor/blob/develop/HISTORY.md)
- [Commits](https://github.com/josdejong/jsoneditor/compare/v9.1.0...v9.1.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump rollup from 2.28.1 to 2.28.2

Bumps [rollup](https://github.com/rollup/rollup) from 2.28.1 to 2.28.2.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v2.28.1...v2.28.2)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #877 from matteobruni/dependabot/npm_and_yarn/dev/eslint-plugin-vue-7.0.0-beta.4

build(deps-dev): bump eslint-plugin-vue from 7.0.0-beta.3 to 7.0.0-beta.4

matteobruni committed:
Merge pull request #878 from matteobruni/dependabot/npm_and_yarn/dev/svelte-preprocess-4.3.2

build(deps-dev): bump svelte-preprocess from 4.3.0 to 4.3.2

matteobruni committed:
Merge pull request #880 from matteobruni/dependabot/npm_and_yarn/dev/jsoneditor-9.1.1

build(deps-dev): bump jsoneditor from 9.1.0 to 9.1.1

matteobruni committed:
Merge pull request #881 from matteobruni/dependabot/npm_and_yarn/dev/rollup-2.28.2

build(deps-dev): bump rollup from 2.28.1 to 2.28.2

dependabot-preview[bot] committed:
build(deps-dev): bump svelte-check from 1.0.49 to 1.0.50

Bumps [svelte-check](https://github.com/sveltejs/language-tools) from 1.0.49 to 1.0.50.
- [Release notes](https://github.com/sveltejs/language-tools/releases)
- [Commits](https://github.com/sveltejs/language-tools/commits)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #879 from matteobruni/dependabot/npm_and_yarn/dev/svelte-check-1.0.50

build(deps-dev): bump svelte-check from 1.0.49 to 1.0.50

matteobruni committed:
feat: changed loadJSON with multiple configs and an optional index to select one of them

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

matteobruni committed:
build: import type fix

matteobruni committed:
build: removed unused spatial grid

matteobruni committed:
refactor: added getRadius method on particles and used where needed

matteobruni committed:
refactor: color and rotate animations are now made with the same logic of size and opacity

matteobruni committed:
refactor: stroke color animation aligned with other like the previous commit

matteobruni committed:
refactor: small fix

matteobruni committed:
refactor: removed unused is overlapping method

matteobruni committed:
build: fixed tests

matteobruni committed:
build: fix editor lint command

matteobruni committed:
build: fixed editor main file

matteobruni committed:
build: fixed editor webpack config

matteobruni committed:
build: updated preact demo file

matteobruni committed:
refactor: fixed some code warnings

matteobruni committed:
build: fixed ssr file

matteobruni committed:
refactor: cleaned code in div bounce handler

matteobruni committed:
refactor: collisions between particles and div now share code

matteobruni committed:
build: cleaned code warning

matteobruni committed:
build: cleaned code warning

matteobruni committed:
refactor: links and grab links color now share the color logic code

matteobruni committed:
fix: fixed rotate following path

matteobruni committed:
feat: added feature from issue #797

matteobruni committed:
build: cleaned code warning

dependabot-preview[bot] committed:
build(deps-dev): bump ng-packagr from 10.1.0 to 10.1.2

Bumps [ng-packagr](https://github.com/ng-packagr/ng-packagr) from 10.1.0 to 10.1.2.
- [Release notes](https://github.com/ng-packagr/ng-packagr/releases)
- [Changelog](https://github.com/ng-packagr/ng-packagr/blob/master/CHANGELOG.md)
- [Commits](https://github.com/ng-packagr/ng-packagr/compare/v10.1.0...v10.1.2)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump uglify-js from 3.10.4 to 3.11.0

Bumps [uglify-js](https://github.com/mishoo/UglifyJS) from 3.10.4 to 3.11.0.
- [Release notes](https://github.com/mishoo/UglifyJS/releases)
- [Commits](https://github.com/mishoo/UglifyJS/compare/v3.10.4...v3.11.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump rollup-plugin-typescript2 from 0.27.2 to 0.27.3

Bumps [rollup-plugin-typescript2](https://github.com/ezolenko/rollup-plugin-typescript2) from 0.27.2 to 0.27.3.
- [Release notes](https://github.com/ezolenko/rollup-plugin-typescript2/releases)
- [Commits](https://github.com/ezolenko/rollup-plugin-typescript2/compare/0.27.2...0.27.3)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump @angular/forms from 10.1.2 to 10.1.3

Bumps [@angular/forms](https://github.com/angular/angular/tree/HEAD/packages/forms) from 10.1.2 to 10.1.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/10.1.3/packages/forms)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #884 from matteobruni/dependabot/npm_and_yarn/dev/ng-packagr-10.1.2

build(deps-dev): bump ng-packagr from 10.1.0 to 10.1.2

matteobruni committed:
Merge pull request #885 from matteobruni/dependabot/npm_and_yarn/dev/uglify-js-3.11.0

build(deps-dev): bump uglify-js from 3.10.4 to 3.11.0

matteobruni committed:
Merge pull request #886 from matteobruni/dependabot/npm_and_yarn/dev/rollup-plugin-typescript2-0.27.3

build(deps-dev): bump rollup-plugin-typescript2 from 0.27.2 to 0.27.3

matteobruni committed:
Merge pull request #887 from matteobruni/dependabot/npm_and_yarn/dev/angular/forms-10.1.3

build(deps): bump @angular/forms from 10.1.2 to 10.1.3

matteobruni committed:
feat: particle with move max distance are now good, closes #740

matteobruni committed:
refactor: initial position variable in mover

matteobruni committed:
build: improved fountain gravity sample

matteobruni committed:
feat: added motion options to handle prefers-reduced-motion media query, fixes #888

matteobruni committed:
fix: fixed prefers reduced motion query

matteobruni committed:
fix: fixed other animations with reduced motion

matteobruni committed:
build: fixed build

matteobruni committed:
build: updated main demo index page

matteobruni committed:
Merge branch 'master' into dev

# Conflicts:
#	.gitignore
#	core/main/package.json
#	yarn.lock

matteobruni committed:
build: updates

matteobruni committed:
feat: added motion options to editor

dependabot-preview[bot] committed:
build(deps-dev): bump @angular-devkit/build-ng-packagr

Bumps [@angular-devkit/build-ng-packagr](https://github.com/angular/angular-cli) from 0.1001.2 to 0.1001.3.
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Commits](https://github.com/angular/angular-cli/commits)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #891 from matteobruni/dependabot/npm_and_yarn/dev/angular-devkit/build-ng-packagr-0.1001.3

build(deps-dev): bump @angular-devkit/build-ng-packagr from 0.1001.2 to 0.1001.3

matteobruni committed:
refactor: motion classes structure fixed, added documentation

matteobruni committed:
docs: added motion inside Options.md

matteobruni committed:
fix: fixed life duration/delay sync options

matteobruni committed:
build: editor version aligned to tsparticles, they must be always synced to avoid confusion

Commit:
feat: add hsv color support

matteobruni committed:
Merge pull request #893 from thecodingdog/dev-Add-hsv-support

feat: add hsv color support

Commit:
feat: hsv color documentation

matteobruni committed:
Merge pull request #894 from thecodingdog/dev-Add-hsv-support

feat: hsv color documentation

matteobruni committed:
refactor: tsParticles library exports the right IOptions based on import and Container type
refactor: all components libraries import/export the same options types, closes #898

matteobruni committed:
refactor: exported the partial options in a definite type

code-factor committed:
[CodeFactor] Apply fixes to commit 2e86b57

[ci skip] [skip ci]

code-factor committed:
[CodeFactor] Apply fixes to commit 2e86b57

[ci skip] [skip ci]

matteobruni committed:
refactor: demos now uses the new options type

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

matteobruni committed:
Merge pull request #899 from matteobruni/dev-cf-autofix

Apply fixes from CodeFactor

matteobruni committed:
refactor: fixed font awesome preset

dependabot-preview[bot] committed:
build(deps): bump @angular/common from 10.1.3 to 10.1.4

Bumps [@angular/common](https://github.com/angular/angular/tree/HEAD/packages/common) from 10.1.3 to 10.1.4.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/10.1.4/packages/common)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump preact from 10.5.2 to 10.5.3

Bumps [preact](https://github.com/preactjs/preact) from 10.5.2 to 10.5.3.
- [Release notes](https://github.com/preactjs/preact/releases)
- [Commits](https://github.com/preactjs/preact/compare/10.5.2...10.5.3)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump postcss from 8.0.9 to 8.1.1

Bumps [postcss](https://github.com/postcss/postcss) from 8.0.9 to 8.1.1.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/master/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.0.9...8.1.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump @fortawesome/fontawesome-free

Bumps [@fortawesome/fontawesome-free](https://github.com/FortAwesome/Font-Awesome) from 5.14.0 to 5.15.0.
- [Release notes](https://github.com/FortAwesome/Font-Awesome/releases)
- [Changelog](https://github.com/FortAwesome/Font-Awesome/blob/master/CHANGELOG.md)
- [Commits](https://github.com/FortAwesome/Font-Awesome/compare/5.14.0...5.15.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump @angular/core from 10.1.3 to 10.1.4

Bumps [@angular/core](https://github.com/angular/angular/tree/HEAD/packages/core) from 10.1.3 to 10.1.4.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/10.1.4/packages/core)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #904 from matteobruni/dependabot/npm_and_yarn/dev/fortawesome/fontawesome-free-5.15.0

build(deps-dev): bump @fortawesome/fontawesome-free from 5.14.0 to 5.15.0

matteobruni committed:
Merge pull request #903 from matteobruni/dependabot/npm_and_yarn/dev/postcss-8.1.1

build(deps-dev): bump postcss from 8.0.9 to 8.1.1

matteobruni committed:
Merge pull request #902 from matteobruni/dependabot/npm_and_yarn/dev/preact-10.5.3

build(deps): bump preact from 10.5.2 to 10.5.3

matteobruni committed:
Merge pull request #905 from matteobruni/dependabot/npm_and_yarn/dev/angular/core-10.1.4

build(deps): bump @angular/core from 10.1.3 to 10.1.4

matteobruni committed:
Merge pull request #901 from matteobruni/dependabot/npm_and_yarn/dev/angular/common-10.1.4

build(deps): bump @angular/common from 10.1.3 to 10.1.4

dependabot-preview[bot] committed:
build(deps): bump @angular/animations from 10.1.3 to 10.1.4

Bumps [@angular/animations](https://github.com/angular/angular/tree/HEAD/packages/animations) from 10.1.3 to 10.1.4.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/10.1.4/packages/animations)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #900 from matteobruni/dependabot/npm_and_yarn/dev/angular/animations-10.1.4

build(deps): bump @angular/animations from 10.1.3 to 10.1.4

matteobruni committed:
Merge branch 'master' into dev

matteobruni committed:
Merge branch 'master' into dev

matteobruni committed:
feat: manual particles, fixes #839

matteobruni committed:
build: updates

matteobruni committed:
build: alpha to vue3 package

matteobruni committed:
chore(release): published new version

 - ng-particles@2.1.0-beta.0
 - inferno-particles@1.1.0-beta.0
 - jquery-particles@1.18.0-beta.0
 - preact-particles@1.18.0-beta.0
 - react-tsparticles@1.18.0-beta.0
 - svelte-particles@1.1.0-beta.0
 - particles.vue@2.1.0-beta.0
 - particles.vue3@1.1.0-beta.0
 - tsparticles-editor@1.18.0-beta.0
 - tsparticles@1.18.0-beta.0
 - ng-particles-demo@1.3.0-beta.0
 - editor-demo@1.0.0-beta.0
 - inferno-particles-demo@1.1.0-beta.0
 - jquery-particles-demo@1.3.0-beta.0
 - tsparticles-demo@1.3.0-beta.0
 - preact-particles-demo@1.3.0-beta.0
 - react-tsparticles-demo@1.3.0-beta.0
 - svelte-demo@1.1.0-beta.0
 - particles.vue-demo@1.3.0-beta.0
 - particles.vue3-demo@1.0.3-beta.0
 - tsparticles-preset-60fps@1.3.0-beta.0
 - tsparticles-preset-background-mask@1.3.0-beta.0
 - tsparticles-preset-basic@1.3.0-beta.0
 - tsparticles-preset-big-circles@1.3.0-beta.0
 - tsparticles-preset-bouncing@1.3.0-beta.0
 - tsparticles-preset-fire@1.3.0-beta.0
 - tsparticles-preset-font-awesome@1.3.0-beta.0
 - tsparticles-preset-fountain@1.3.0-beta.0
 - tsparticles-preset-snow@1.3.0-beta.0
 - tsparticles-preset-stars@1.3.0-beta.0
 - tsparticles-shape-bubble@1.3.0-beta.0
 - tsparticles-shape-heart@1.3.0-beta.0
 - tsparticles-shape-multiline-text@1.3.0-beta.0
 - tsparticles-shape-spiral@1.3.0-beta.0
 - cra-template-particles-typescript@1.1.0-beta.0
 - cra-template-particles@1.1.0-beta.0

matteobruni committed:
build: update editor demo version update

matteobruni committed:
build: version updates

matteobruni committed:
chore(release): published new version

 - editor-demo@1.0.0-beta.8

matteobruni committed:
build: updated editor version

matteobruni committed:
chore(release): published new version

 - editor-demo@1.0.0-beta.10

matteobruni committed:
build: editor demo version update

matteobruni committed:
chore(release): published new version

 - editor-demo@1.0.0-beta.31

matteobruni committed:
build: fixed vue3 version

matteobruni committed:
build: fixed packages.json

matteobruni committed:
chore(release): published new version

 - inferno-particles@1.1.0-beta.1
 - tsparticles-editor@1.18.0-beta.1
 - editor-demo@1.0.0-beta.32
 - inferno-particles-demo@1.1.0-beta.1
 - particles.vue3-demo@1.0.3-beta.1
 - cra-template-particles-typescript@1.1.0-beta.1
 - cra-template-particles@1.1.0-beta.1

dependabot-preview[bot] committed:
build(deps): bump @angular/router from 10.1.3 to 10.1.4

Bumps [@angular/router](https://github.com/angular/angular/tree/HEAD/packages/router) from 10.1.3 to 10.1.4.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/10.1.4/packages/router)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #908 from matteobruni/dependabot/npm_and_yarn/dev/angular/router-10.1.4

build(deps): bump @angular/router from 10.1.3 to 10.1.4

matteobruni committed:
Merge pull request #740 from matteobruni/dev

1.18.0

ghost commented:
## DeepCode's analysis on [#62eaa1](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/179845c0ffad7bf3c6bb063864363148c6dd6f65/matteobruni/tsparticles/62eaa1d3c7cda8682663e85ce92167d3163b763e/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=0&) found:

- :warning: **1** warning :point_down:
- :heavy_check_mark: **3** issues were fixed.
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">Using component state to compute the next state could lead to undesired behavior, since state updates may be batched and run asynchronously by React. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/62eaa1d3c7cda8682663e85ce92167d3163b763e/components/preact/src/Particles.tsx#L35">Particles.tsx:35</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/62eaa1d3c7cda8682663e85ce92167d3163b763e/components/preact/src/Particles.tsx#L45">Particles.tsx:45</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/62eaa1d3c7cda8682663e85ce92167d3163b763e/components/preact/src/Particles.tsx#L80">Particles.tsx:80</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/179845c0ffad7bf3c6bb063864363148c6dd6f65/matteobruni/tsparticles/62eaa1d3c7cda8682663e85ce92167d3163b763e/pr/_/%2Fcomponents%2Fpreact%2Fsrc%2FParticles.tsx/javascript%2Fdc_interfile_project%2FReact-stateUsedInStateUpdateMethod/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/179845c0ffad7bf3c6bb063864363148c6dd6f65/matteobruni/tsparticles/62eaa1d3c7cda8682663e85ce92167d3163b763e/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=0&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=matteobruni)

matteobruni committed:
feat: added bounce to mouse hover event

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

matteobruni committed:
chore(release): published new version

 - ng-particles@2.1.0-beta.1
 - inferno-particles@1.1.0-beta.2
 - jquery-particles@1.18.0-beta.1
 - preact-particles@1.18.0-beta.1
 - react-tsparticles@1.18.0-beta.1
 - svelte-particles@1.1.0-beta.1
 - particles.vue@2.1.0-beta.1
 - particles.vue3@1.1.0-beta.1
 - tsparticles-editor@1.18.0-beta.2
 - tsparticles@1.18.0-beta.1
 - ng-particles-demo@1.3.0-beta.1
 - editor-demo@1.0.0-beta.33
 - inferno-particles-demo@1.1.0-beta.2
 - jquery-particles-demo@1.3.0-beta.1
 - tsparticles-demo@1.3.0-beta.1
 - preact-particles-demo@1.3.0-beta.1
 - react-tsparticles-demo@1.3.0-beta.1
 - svelte-demo@1.1.0-beta.1
 - particles.vue-demo@1.3.0-beta.1
 - particles.vue3-demo@1.0.3-beta.2
 - tsparticles-preset-60fps@1.3.0-beta.1
 - tsparticles-preset-background-mask@1.3.0-beta.1
 - tsparticles-preset-basic@1.3.0-beta.1
 - tsparticles-preset-big-circles@1.3.0-beta.1
 - tsparticles-preset-bouncing@1.3.0-beta.1
 - tsparticles-preset-fire@1.3.0-beta.1
 - tsparticles-preset-font-awesome@1.3.0-beta.1
 - tsparticles-preset-fountain@1.3.0-beta.1
 - tsparticles-preset-snow@1.3.0-beta.1
 - tsparticles-preset-stars@1.3.0-beta.1
 - tsparticles-shape-bubble@1.3.0-beta.1
 - tsparticles-shape-heart@1.3.0-beta.1
 - tsparticles-shape-multiline-text@1.3.0-beta.1
 - tsparticles-shape-spiral@1.3.0-beta.1
 - cra-template-particles-typescript@1.1.0-beta.2
 - cra-template-particles@1.1.0-beta.2

jpyamamoto committed:
Spanish translation of main readme and some docs

matteobruni committed:
Merge pull request #911 from JPYamamoto/spanish-translation

Spanish translation of main readme and some docs

matteobruni committed:
fix: fixed issue with new bounce active conditions

matteobruni committed:
chore(release): published new version

 - ng-particles@2.1.0-beta.2
 - inferno-particles@1.1.0-beta.3
 - jquery-particles@1.18.0-beta.2
 - preact-particles@1.18.0-beta.2
 - react-tsparticles@1.18.0-beta.2
 - svelte-particles@1.1.0-beta.2
 - particles.vue@2.1.0-beta.2
 - particles.vue3@1.1.0-beta.2
 - tsparticles-editor@1.18.0-beta.3
 - tsparticles@1.18.0-beta.2
 - ng-particles-demo@1.3.0-beta.2
 - editor-demo@1.0.0-beta.34
 - inferno-particles-demo@1.1.0-beta.3
 - jquery-particles-demo@1.3.0-beta.2
 - tsparticles-demo@1.3.0-beta.2
 - preact-particles-demo@1.3.0-beta.2
 - react-tsparticles-demo@1.3.0-beta.2
 - svelte-demo@1.1.0-beta.2
 - particles.vue-demo@1.3.0-beta.2
 - particles.vue3-demo@1.0.3-beta.3
 - tsparticles-preset-60fps@1.3.0-beta.2
 - tsparticles-preset-background-mask@1.3.0-beta.2
 - tsparticles-preset-basic@1.3.0-beta.2
 - tsparticles-preset-big-circles@1.3.0-beta.2
 - tsparticles-preset-bouncing@1.3.0-beta.2
 - tsparticles-preset-fire@1.3.0-beta.2
 - tsparticles-preset-font-awesome@1.3.0-beta.2
 - tsparticles-preset-fountain@1.3.0-beta.2
 - tsparticles-preset-snow@1.3.0-beta.2
 - tsparticles-preset-stars@1.3.0-beta.2
 - tsparticles-shape-bubble@1.3.0-beta.2
 - tsparticles-shape-heart@1.3.0-beta.2
 - tsparticles-shape-multiline-text@1.3.0-beta.2
 - tsparticles-shape-spiral@1.3.0-beta.2
 - cra-template-particles-typescript@1.1.0-beta.3
 - cra-template-particles@1.1.0-beta.3

matteobruni committed:
build: code update to Bouncer.ts

matteobruni committed:
Merge pull request #910 from matteobruni/dev

1.18.0-beta.2

matteobruni committed:
Merge branch 'master' into dev

matteobruni committed:
Merge pull request #913 from matteobruni/dev

Updating staging with latest commits

matteobruni committed:
fix: fixed rotation animation issues

matteobruni committed:
chore(release): published new version

 - ng-particles@2.1.0-beta.3
 - inferno-particles@1.1.0-beta.4
 - jquery-particles@1.18.0-beta.3
 - preact-particles@1.18.0-beta.3
 - react-tsparticles@1.18.0-beta.3
 - svelte-particles@1.1.0-beta.3
 - particles.vue@2.1.0-beta.3
 - particles.vue3@1.1.0-beta.3
 - tsparticles-editor@1.18.0-beta.4
 - tsparticles@1.18.0-beta.3
 - ng-particles-demo@1.3.0-beta.3
 - editor-demo@1.0.0-beta.35
 - inferno-particles-demo@1.1.0-beta.4
 - jquery-particles-demo@1.3.0-beta.3
 - tsparticles-demo@1.3.0-beta.3
 - preact-particles-demo@1.3.0-beta.3
 - react-tsparticles-demo@1.3.0-beta.3
 - svelte-demo@1.1.0-beta.3
 - particles.vue-demo@1.3.0-beta.3
 - particles.vue3-demo@1.0.3-beta.4
 - tsparticles-preset-60fps@1.3.0-beta.3
 - tsparticles-preset-background-mask@1.3.0-beta.3
 - tsparticles-preset-basic@1.3.0-beta.3
 - tsparticles-preset-big-circles@1.3.0-beta.3
 - tsparticles-preset-bouncing@1.3.0-beta.3
 - tsparticles-preset-fire@1.3.0-beta.3
 - tsparticles-preset-font-awesome@1.3.0-beta.3
 - tsparticles-preset-fountain@1.3.0-beta.3
 - tsparticles-preset-snow@1.3.0-beta.3
 - tsparticles-preset-stars@1.3.0-beta.3
 - tsparticles-shape-bubble@1.3.0-beta.3
 - tsparticles-shape-heart@1.3.0-beta.3
 - tsparticles-shape-multiline-text@1.3.0-beta.3
 - tsparticles-shape-spiral@1.3.0-beta.3
 - cra-template-particles-typescript@1.1.0-beta.4
 - cra-template-particles@1.1.0-beta.4

matteobruni committed:
Merge pull request #914 from matteobruni/dev

1.18.0-beta.3

matteobruni committed:
fix: fixed bounce on rectangular divs

matteobruni committed:
chore(release): published new version

 - ng-particles@2.1.0-beta.4
 - inferno-particles@1.1.0-beta.5
 - jquery-particles@1.18.0-beta.4
 - preact-particles@1.18.0-beta.4
 - react-tsparticles@1.18.0-beta.4
 - svelte-particles@1.1.0-beta.4
 - particles.vue@2.1.0-beta.4
 - particles.vue3@1.1.0-beta.4
 - tsparticles-editor@1.18.0-beta.5
 - tsparticles@1.18.0-beta.4
 - ng-particles-demo@1.3.0-beta.4
 - editor-demo@1.0.0-beta.36
 - inferno-particles-demo@1.1.0-beta.5
 - jquery-particles-demo@1.3.0-beta.4
 - tsparticles-demo@1.3.0-beta.4
 - preact-particles-demo@1.3.0-beta.4
 - react-tsparticles-demo@1.3.0-beta.4
 - svelte-demo@1.1.0-beta.4
 - particles.vue-demo@1.3.0-beta.4
 - particles.vue3-demo@1.0.3-beta.5
 - tsparticles-preset-60fps@1.3.0-beta.4
 - tsparticles-preset-background-mask@1.3.0-beta.4
 - tsparticles-preset-basic@1.3.0-beta.4
 - tsparticles-preset-big-circles@1.3.0-beta.4
 - tsparticles-preset-bouncing@1.3.0-beta.4
 - tsparticles-preset-fire@1.3.0-beta.4
 - tsparticles-preset-font-awesome@1.3.0-beta.4
 - tsparticles-preset-fountain@1.3.0-beta.4
 - tsparticles-preset-snow@1.3.0-beta.4
 - tsparticles-preset-stars@1.3.0-beta.4
 - tsparticles-shape-bubble@1.3.0-beta.4
 - tsparticles-shape-heart@1.3.0-beta.4
 - tsparticles-shape-multiline-text@1.3.0-beta.4
 - tsparticles-shape-spiral@1.3.0-beta.4
 - cra-template-particles-typescript@1.1.0-beta.5
 - cra-template-particles@1.1.0-beta.5

matteobruni committed:
build: prettier

matteobruni committed:
Merge pull request #915 from matteobruni/dev

1.18.0-beta.4

matteobruni committed:
Merge branch 'master' into dev

matteobruni committed:
build: updated deps

matteobruni committed:
build: updated deps

matteobruni committed:
build: updated deps

matteobruni committed:
build: prettier

matteobruni committed:
chore(release): published new version

 - ng-particles@2.1.0-beta.5
 - inferno-particles@1.1.0-beta.6
 - jquery-particles@1.18.0-beta.5
 - preact-particles@1.18.0-beta.5
 - react-tsparticles@1.18.0-beta.5
 - svelte-particles@1.1.0-beta.5
 - particles.vue@2.1.0-beta.5
 - particles.vue3@1.1.0-beta.5
 - tsparticles-editor@1.18.0-beta.6
 - tsparticles@1.18.0-beta.5
 - ng-particles-demo@1.3.0-beta.5
 - editor-demo@1.0.0-beta.37
 - inferno-particles-demo@1.1.0-beta.6
 - jquery-particles-demo@1.3.0-beta.5
 - tsparticles-demo@1.3.0-beta.5
 - preact-particles-demo@1.3.0-beta.5
 - react-tsparticles-demo@1.3.0-beta.5
 - svelte-demo@1.1.0-beta.5
 - particles.vue-demo@1.3.0-beta.5
 - particles.vue3-demo@1.0.3-beta.6
 - tsparticles-preset-60fps@1.3.0-beta.5
 - tsparticles-preset-background-mask@1.3.0-beta.5
 - tsparticles-preset-basic@1.3.0-beta.5
 - tsparticles-preset-big-circles@1.3.0-beta.5
 - tsparticles-preset-bouncing@1.3.0-beta.5
 - tsparticles-preset-fire@1.3.0-beta.5
 - tsparticles-preset-font-awesome@1.3.0-beta.5
 - tsparticles-preset-fountain@1.3.0-beta.5
 - tsparticles-preset-snow@1.3.0-beta.5
 - tsparticles-preset-stars@1.3.0-beta.5
 - tsparticles-shape-bubble@1.3.0-beta.5
 - tsparticles-shape-heart@1.3.0-beta.5
 - tsparticles-shape-multiline-text@1.3.0-beta.5
 - tsparticles-shape-spiral@1.3.0-beta.5
 - cra-template-particles-typescript@1.1.0-beta.6
 - cra-template-particles@1.1.0-beta.6

matteobruni committed:
Merge pull request #917 from matteobruni/dev

1.18.0-beta.5

dependabot-preview[bot] committed:
build(deps-dev): bump eslint-plugin-vue from 7.0.0 to 7.0.1

Bumps [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) from 7.0.0 to 7.0.1.
- [Release notes](https://github.com/vuejs/eslint-plugin-vue/releases)
- [Commits](https://github.com/vuejs/eslint-plugin-vue/compare/v7.0.0...v7.0.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps-dev): bump uglify-js from 3.11.0 to 3.11.1

Bumps [uglify-js](https://github.com/mishoo/UglifyJS) from 3.11.0 to 3.11.1.
- [Release notes](https://github.com/mishoo/UglifyJS/releases)
- [Commits](https://github.com/mishoo/UglifyJS/compare/v3.11.0...v3.11.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

matteobruni committed:
Merge pull request #918 from matteobruni/dependabot/npm_and_yarn/dev/eslint-plugin-vue-7.0.1

build(deps-dev): bump eslint-plugin-vue from 7.0.0 to 7.0.1

matteobruni committed:
Merge pull request #920 from matteobruni/dependabot/npm_and_yarn/dev/uglify-js-3.11.1

build(deps-dev): bump uglify-js from 3.11.0 to 3.11.1

matteobruni committed:
build: editor demo version update

matteobruni committed:
chore(release): published new version

 - ng-particles@2.1.0
 - inferno-particles@1.1.0
 - jquery-particles@1.18.0
 - preact-particles@1.18.0
 - react-tsparticles@1.18.0
 - svelte-particles@1.1.0
 - particles.vue@2.1.0
 - particles.vue3@1.1.0
 - tsparticles-editor@1.18.0
 - tsparticles@1.18.0
 - ng-particles-demo@1.3.0
 - editor-demo@1.3.0
 - inferno-particles-demo@1.1.0
 - jquery-particles-demo@1.3.0
 - tsparticles-demo@1.3.0
 - preact-particles-demo@1.3.0
 - react-tsparticles-demo@1.3.0
 - svelte-demo@1.1.0
 - particles.vue-demo@1.3.0
 - particles.vue3-demo@1.1.0
 - tsparticles-preset-60fps@1.3.0
 - tsparticles-preset-background-mask@1.3.0
 - tsparticles-preset-basic@1.3.0
 - tsparticles-preset-big-circles@1.3.0
 - tsparticles-preset-bouncing@1.3.0
 - tsparticles-preset-fire@1.3.0
 - tsparticles-preset-font-awesome@1.3.0
 - tsparticles-preset-fountain@1.3.0
 - tsparticles-preset-snow@1.3.0
 - tsparticles-preset-stars@1.3.0
 - tsparticles-shape-bubble@1.3.0
 - tsparticles-shape-heart@1.3.0
 - tsparticles-shape-multiline-text@1.3.0
 - tsparticles-shape-spiral@1.3.0
 - cra-template-particles-typescript@1.1.0
 - cra-template-particles@1.1.0

matteobruni committed:
Merge pull request #923 from matteobruni/dev

v1.18.0
